### PR TITLE
Inconsistent Keys

### PR DESCRIPTION
**Pull request by DrewLazzeriKitware**
# Changes
1) Adding yaml to fix keys which could not be set
2) Adding yaml to fix keys which could only be set in a certain order
3) Adding a new domain to validate that a list is a subset of another list
4) Adding a missing key (`PhaseSources.GeomNames`)

# Testing 
I have tests which I need to integrate with the testing framework here, but would like to submit this PR for approval first.

### CODE REVIEW

File: pf-keys/definitions/geom.yaml [hunk 1]
```diff
@@ -354,7 +354,13 @@ Geom:
...
         entire computational domain.
       domains:
-        AnyString:
+        EnumDomain:
+          locations: 
+            - /GeomInput/{GeomInputItem}/GeomNames
+            - /GeomInput/{GeomInputItem}/GeomName
+      handlers:
+        NamesToListConverter:
+          type: EnumLocationsHandler
```
Comment (jourdain) [inline pf-keys/definitions/geom.yaml]: Weird name for an handler...

File: pftools/python/parflow/tools/database/handlers.py [hunk 2]
```diff
@@ -97,6 +101,17 @@ def decorate(self, value, container, class_name=None, location='.',
         raise ValueHandlerException(
             f'{value} is not of the expected type for ChildrenHandler')
 
+# -----------------------------------------------------------------------------
+class EnumLocationsHandler:
```
Comment (jourdain) [inline pftools/python/parflow/tools/database/handlers.py]: I guess they are some confusion on what is a `handler` and a `domain` is. Basically you can not be both hence the name of that class does not make any sense.

Comment (jourdain) [reply pftools/python/parflow/tools/database/handlers.py]: The proper name should be `ListHandler`.

The responsibility of that handler is to ensure the value of that property get converted into a list. What you do with that list should be irrelevant to the handler...

File: pf-keys/definitions/phase.yaml [hunk 3]
```diff
@@ -50,7 +50,14 @@ Phase:
...
         location: /Geom/{GeomItem}/ICSaturation
-
+      BCSaturationPhaseUpdater:
+        type: ChildrenHandler
+        class_name: BCSatPhaseItem
+        location: /Patch/{BCItem}/BCSaturation
+      BCPressureIntervalPhaseUpdater:
+        type: ChildrenHandler
+        class_name: BCPressIntPhaseItem
+        location: /Patch/{BCItem}/BCPressure/{BCPressureIntervalItem}
```
Comment (jourdain) [inline pf-keys/definitions/phase.yaml:60]: Does `/Patch/{BCItem}/BCPressure/{BCPressureIntervalItem}/{BCPressIntPhaseItem}` has a __from__ ?

File: pf-keys/definitions/phase.yaml [hunk 4]
```diff
@@ -526,6 +533,10 @@ Contaminants:
         type: ChildrenHandler
         class_name: GeomContItem
         location: /Geom/{GeomItem}/
+      ConcentrationPhaseContaminantUpdater:
+        type: ChildrenHandler
+        class_name: ConcentrationPhaseContItem
+        location: /Wells/{WellItem}/{WellIntervalItem}/Concentration/{ConcentrationPhaseItem}
```
Comment (jourdain) [inline pf-keys/definitions/phase.yaml:576]: Does `/Wells/{WellItem}/{WellIntervalItem}/Concentration/{ConcentrationPhaseItem}/{ConcentrationPhaseContItem}` has a __from__?

Comment (DrewLazzeriKitware) [reply pf-keys/definitions/phase.yaml:576]: Yes, here https://github.com/DrewLazzeriKitware/parflow/blob/b4c9ec204d1168854138f816af185ef719b79028/pf-keys/definitions/wells.yaml#L119

File: pf-keys/definitions/phase.yaml (hunk 3 above)
Comment (DrewLazzeriKitware) [reply pf-keys/definitions/phase.yaml:60]: Yes, here https://github.com/DrewLazzeriKitware/parflow/blob/b4c9ec204d1168854138f816af185ef719b79028/pf-keys/definitions/bconditions.yaml#L205

**Pull request merged**